### PR TITLE
unconditionally enable Blending

### PR DESCRIPTION
**Pull request by atupone**
Remove blend from the option menu.
Consider blend always true in the code.

### CODE REVIEW

File: include/SceneNode.h [hunk 1]
```diff
@@ -51,7 +51,6 @@
 #define myColor4f(r, g, b, a)   SceneNode::glColor4f(r, g, b, a)
 #define myColor3fv(rgb)     SceneNode::glColor3fv(rgb)
 #define myColor4fv(rgba)    SceneNode::glColor4fv(rgba)
-#define myStipple(alpha)    SceneNode::setStipple(alpha)
```
Comment (jwmelto) [inline include/SceneNode.h]: stipple wasn't removed above; are you sure you meant to do this?

File: src/bzflag/DisplayMenu.cxx [hunk 2]
```diff
@@ -404,10 +393,6 @@ void            DisplayMenu::callback(HUDuiControl* w, const void* data)
         BZDB.set("dither", list->getIndex() ? "1" : "0");
         sceneRenderer->notifyStyleChange();
         break;
-    case '2':
```
Comment (jwmelto) [inline src/bzflag/DisplayMenu.cxx]: you shouldn't just remove case 2. Sadly, there is no default case in this switch. Either add a default case (which is best) or an empty or error case for 2.

File: src/bzflag/playing.cxx [hunk 3]
```diff
@@ -3625,7 +3625,7 @@ bool            addExplosion(const float* _pos,
     if (sceneRenderer->useQuality() < 1) return false;
 
     // don't add explosion if blending or texture mapping are off
```
Comment (jwmelto) [inline src/bzflag/playing.cxx]: change the comment to reflect the code.

File: src/geometry/BoltSceneNode.cxx [hunk 4]
```diff
@@ -159,33 +159,23 @@ void            BoltSceneNode::addLight(
...
-        builder.setAlphaFunc();
-        if ((RENDERER.useQuality() >= 3) && drawFlares)
-        {
-            builder.setShading(GL_SMOOTH);
-            builder.enableMaterial(false);
-        }
-        else
-            builder.setShading(texturing ? GL_FLAT : GL_SMOOTH);
-    }
+    const int shotLength = (int)(BZDBCache::shotLength * 3.0f);
```
Comment (jwmelto) [inline src/geometry/BoltSceneNode.cxx]: C-style casts are evil. Either use a C++ cast or a "function-style" cast (anonymous temporary:
```c++
const int shotLength = static_cast<int>(BZDBCache::shotLength * 3.0);
const int shotLength = int( BZDBCache::shotLength * 3.0 );
```
(although I don't know why we are multiplying by a float anyway)
(and magic numbers like "3.0" are maintenance burdens)
(and I realize you just removed the `if` but any time you touch a line of code, it's nice to leave it better)

File: include/SceneNode.h (hunk 1 above)
Comment (atupone) [reply include/SceneNode.h]: setStipple is used, myStipple is no more

File: src/bzflag/DisplayMenu.cxx (hunk 2 above)
Comment (atupone) [reply src/bzflag/DisplayMenu.cxx]: I agree! Doing it

File: src/bzflag/playing.cxx (hunk 3 above)
Comment (atupone) [reply src/bzflag/playing.cxx]: OK

File: src/geometry/BoltSceneNode.cxx (hunk 4 above)
Comment (atupone) [reply src/geometry/BoltSceneNode.cxx]: I don't know why we multiply it either.
Removing

Comment (jwmelto) [reply src/geometry/BoltSceneNode.cxx]: I wasn’t suggesting the multiplication was bad, just that multiplying by `3.0f` vs `3` was silly, especially given the conversion to `int`

Comment (atupone) [reply src/geometry/BoltSceneNode.cxx]: I re-added it

Comment (macsforme) [reply src/geometry/BoltSceneNode.cxx]: > I wasn’t suggesting the multiplication was bad, just that multiplying by `3.0f` vs `3` was silly, especially given the conversion to `int`

I just wanted to point out that the result is not necessarily the same (if the value has a fractional part).

Comment (jwmelto) [reply src/geometry/BoltSceneNode.cxx]: Not to quibble but it is the same. `1.1 * 3.0 == 1.1 * 3`. Integer and floating point numeric operations are well defined and well behaved

Comment (macsforme) [reply src/geometry/BoltSceneNode.cxx]: Okay I see what you were saying. I thought you were suggesting to do the multiplication after the cast.

**Pull request merged**